### PR TITLE
Refactor(eos_designs): Move custom_structured_configuration to python

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Move custom_structured_configuration to python

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

- Render `custom_structured_configuration` and various `structured_config` keys in python instead of jinja2
- remove `struct_cfg` keys from generated structured configuration.

## How to test

Rerun molecule. Only diff is removing `struct_cfg` keys which we don't want in there (since they don't conform to the schema).

Also tested with large customer repo. Runtime for structured_config without PR: 216 seconds. With PR: 88 seconds !!

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Review by ankudinov — APPROVED** [review]
LGTM

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/custom_structured_configuration/avdstructuredconfig.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+from ansible_collections.arista.avd.plugins.plugin_utils.avdfacts import AvdFacts
+from ansible_collections.arista.avd.plugins.plugin_utils.utils import get
+
+
+class AvdStructuredConfig(
+    AvdFacts,
+):
+    """
+    The AvdStructuredConfig Class is imported used "yaml_templates_to_facts" to render parts of the structured config.
+
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/custom_structured_configuration/avdstructuredconfig.py]:
```suggestion
    The AvdStructuredConfig Class is imported using "yaml_templates_to_facts" to render parts of the structured config.

```

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/custom_structured_configuration/avdstructuredconfig.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        if not isinstance(custom_structured_configuration_prefix, list):
+            return [custom_structured_configuration_prefix]
+
+        return custom_structured_configuration_prefix
+
+    @cached_property
+    def _router_bgp(self) -> dict | None:
+        return get(self._hostvars, "router_bgp")
+
+    def _extract_struct_cfg_from_dict_of_dicts(self, data: dict, varname: str) -> dict:
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/custom_structured_configuration/avdstructuredconfig.py]:
```suggestion
    def _extract_struct_cfg_from_dict_of_dicts(self, data: dict, varname: str) -> dict | None:
```

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/custom_structured_configuration/avdstructuredconfig.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        if self._router_bgp is None:
+            return None
+
+        if (struct_cfgs := self._extract_struct_cfg_from_dict_of_dicts(self._router_bgp, "vlans")) is None:
+            return None
+
+        return {"router_bgp": struct_cfgs}
+
+    def _custom_structured_configurations(self) -> list[dict]:
+        EXEMPT_KEYS = ["custom_structured_configuration_prefix", "custom_structured_configuration_list_merge"]
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/custom_structured_configuration/avdstructuredconfig.py]: Even if it is used only here let's move this to the top of the file as a Global Variable (even better would be to read it from some kind of configuration eventually)

**Review by carlbuchmann — APPROVED** [review]
LGTM - successfully tested with two customer inventories.

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/custom_structured_configuration/avdstructuredconfig.py [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+class AvdStructuredConfig(
+    AvdFacts,
+):
+    """
+    The AvdStructuredConfig Class is imported used "yaml_templates_to_facts" to render parts of the structured config.
+
+    "yaml_templates_to_facts" imports, instantiates and run the .render() method on the class.
+
+    The Class uses AvdFacts, as the base class, to get the render, keys and other attributes.
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/custom_structured_configuration/avdstructuredconfig.py]: render is actually coming from this class

**Review by gmuloc — APPROVED** [review]
Tested on project repo using quite a lot of custom_structured_configuration - no change to target confg

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/custom_structured_configuration/__init__.py [hunk 5]
```diff
@@ -0,0 +1,3 @@
+from .avdstructuredconfig import AvdStructuredConfig
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/custom_structured_configuration/__init__.py:1]: I am confused why it is `.avdstructuredconfig`. I haven't used this feature before. So just curious to know.

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/custom_structured_configuration/__init__.py:1]:
```
However, explicit relative imports are an acceptable alternative to absolute imports, especially when dealing with complex package layouts where using absolute imports would be unnecessarily verbose:

from . import sibling
from .sibling import example
```

From: https://peps.python.org/pep-0008/#imports

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/custom_structured_configuration/avdstructuredconfig.py (hunk 1 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/custom_structured_configuration/avdstructuredconfig.py]: updated to "by"

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/custom_structured_configuration/avdstructuredconfig.py (hunk 4 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/custom_structured_configuration/avdstructuredconfig.py]: updated

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/custom_structured_configuration/avdstructuredconfig.py (hunk 2 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/custom_structured_configuration/avdstructuredconfig.py]: Fixed

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/custom_structured_configuration/avdstructuredconfig.py (hunk 3 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/custom_structured_configuration/avdstructuredconfig.py]: done

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/custom_structured_configuration/__init__.py (hunk 5 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/custom_structured_configuration/__init__.py:1]: @tgodaA I will merge this, and we can discuss offline.

**Pull request merged**